Valhalla does not build on macOS with clang 14.0.0

On commit d4c8c1ab39fcbcad91e54160692c0f0aa9ad36aa:

```
➜  build git:(master) clang --version
Apple clang version 14.0.0 (clang-1400.0.29.202)
Target: x86_64-apple-darwin22.1.0
Thread model: posix
InstalledDir: /Library/Developer/CommandLineTools/usr/bin
```

```
➜  build git:(master) make
[  0%] Building CXX object src/midgard/CMakeFiles/valhalla-midgard.dir/logging.cc.o
/Users/dillona/dev/valhalla/src/midgard/logging.cc:42:3: error: 'sprintf' is deprecated: This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead. [-Werror,-Wdeprecated-declarations]
  sprintf(&buffer.front(), "%04d/%02d/%02d %02d:%02d:%09.6f", gmt.tm_year + 1900, gmt.tm_mon + 1,
  ^
/Library/Developer/CommandLineTools/SDKs/MacOSX13.0.sdk/usr/include/stdio.h:188:1: note: 'sprintf' has been explicitly marked deprecated here
__deprecated_msg("This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead.")
^
/Library/Developer/CommandLineTools/SDKs/MacOSX13.0.sdk/usr/include/sys/cdefs.h:215:48: note: expanded from macro '__deprecated_msg'
        #define __deprecated_msg(_msg) __attribute__((__deprecated__(_msg)))
                                                      ^
1 error generated.
make[2]: *** [src/midgard/CMakeFiles/valhalla-midgard.dir/logging.cc.o] Error 1
make[1]: *** [src/midgard/CMakeFiles/valhalla-midgard.dir/all] Error 2
make: *** [all] Error 2
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -3,6 +3,7 @@
 * **Bug Fix**
    * FIXED: valhalla_run_route was missing config logic.[#3824](https://github.com/valhalla/valhalla/pull/3824)
    * FIXED: Handle hexlifying strings with unsigned chars [#3842](https://github.com/valhalla/valhalla/pull/3842)
+   * FIXED: Valhalla does not build on macOS with clang 14.0.0 [#3846](https://github.com/valhalla/valhalla/issues/3846)
 * **Enhancement**
    * ADDED: Add time info to sources_to_targets [#3795](https://github.com/valhalla/valhalla/pull/3795)
    * ADDED: "available_actions" to the /status response [#3836](https://github.com/valhalla/valhalla/pull/3836)

--- a/src/midgard/logging.cc
+++ b/src/midgard/logging.cc
@@ -39,8 +39,8 @@ std::string TimeStamp() {
       (tp - std::chrono::system_clock::from_time_t(tt)) + std::chrono::seconds(gmt.tm_sec);
   // format the string
   std::string buffer("year/mo/dy hr:mn:sc.xxxxxx");
-  sprintf(&buffer.front(), "%04d/%02d/%02d %02d:%02d:%09.6f", gmt.tm_year + 1900, gmt.tm_mon + 1,
-          gmt.tm_mday, gmt.tm_hour, gmt.tm_min, fractional_seconds.count());
+  snprintf(&buffer.front(), buffer.capacity(), "%04d/%02d/%02d %02d:%02d:%09.6f", gmt.tm_year + 1900,
+           gmt.tm_mon + 1, gmt.tm_mday, gmt.tm_hour, gmt.tm_min, fractional_seconds.count());
   return buffer;
 }
 

--- a/src/midgard/logging.cc
+++ b/src/midgard/logging.cc
@@ -39,7 +39,7 @@ std::string TimeStamp() {
       (tp - std::chrono::system_clock::from_time_t(tt)) + std::chrono::seconds(gmt.tm_sec);
   // format the string
   std::string buffer("year/mo/dy hr:mn:sc.xxxxxx");
-  snprintf(&buffer.front(), buffer.capacity(), "%04d/%02d/%02d %02d:%02d:%09.6f", gmt.tm_year + 1900,
+  snprintf(&buffer.front(), buffer.length(), "%04d/%02d/%02d %02d:%02d:%09.6f", gmt.tm_year + 1900,
            gmt.tm_mon + 1, gmt.tm_mday, gmt.tm_hour, gmt.tm_min, fractional_seconds.count());
   return buffer;
 }

